source/application/transforms/react/index.js

```diff
@@ -65,17 +56,19 @@ function convertCode(file, resolveDependencies, opts) {
 	return transform(source, merge({externalHelpers: resolveDependencies}, localOpts)).code;
 }
 
-function createWrappedRenderFunction(file, source, opts) {
+function createWrappedRenderFunction(file, source, resolveDependencies, opts) {
 	let patternJson = loadPatternJson(file.path);
-	let dependencies = writeDependencyImports(file).join('\n');
+	let dependencies = writeDependencyImports(file, resolveDependencies).join('\n');
 	return renderCodeTemplate(source, dependencies, template, pascalCase(patternJson.name), opts);
 }
 
-function writeDependencyImports(file) {
-	let patterns = loadPatternJson(file.path).patterns || {};
-	let dependencies = [];
-	for (let name of Object.keys(file.dependencies)) {
-		dependencies.push(`import ${pascalCase(name)} from '${patterns[name] || name}';`);
+function writeDependencyImports(file, resolveDependencies) {
+	const patterns = loadPatternJson(file.path).patterns || {};
+	const dependencies = [];
+
+	for (const localName of Object.keys(file.dependencies)) {
+		const name = resolveDependencies ? patterns[localName] || localName : localName;
```

Comment:
I think you should add parenthesis here around 'patterns[localName] || localName' to make the operator precedence more clear, even if not required.